packages/framework/ai-collab/src/explicit-strategy/index.ts

```diff
@@ -0,0 +1,362 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	getSimpleSchema,
+	normalizeFieldSchema,
+	Tree,
+	type ImplicitFieldSchema,
+	type SimpleTreeSchema,
+	type TreeNode,
+	type TreeView,
+} from "@fluidframework/tree/internal";
+// eslint-disable-next-line import/no-internal-modules
+import { zodResponseFormat } from "openai/helpers/zod";
+import type {
+	ChatCompletionCreateParams,
+	// eslint-disable-next-line import/no-internal-modules
+} from "openai/resources/index.mjs";
+import { z } from "zod";
+
+import type { OpenAiClientOptions, TokenUsage } from "../aiCollabApi.js";
+
+import { applyAgentEdit } from "./agentEditReducer.js";
+import type { EditWrapper, TreeEdit } from "./agentEditTypes.js";
+import { IdGenerator } from "./idGenerator.js";
+import {
+	getEditingSystemPrompt,
+	getPlanningSystemPrompt,
+	getReviewSystemPrompt,
+	toDecoratedJson,
+	type EditLog,
+} from "./promptGeneration.js";
+import { generateGenericEditTypes } from "./typeGeneration.js";
+import { fail } from "./utils.js";
+
+const DEBUG_LOG: string[] = [];
+
+/**
+ * {@link generateTreeEdits} options.
+ *
+ * @internal
+ */
+export interface GenerateTreeEditsOptions<TSchema extends ImplicitFieldSchema> {
+	openAI: OpenAiClientOptions;
+	treeView: TreeView<TSchema>;
+	treeNode: TreeNode;
+	prompt: {
+		systemRoleContext: string;
+		userAsk: string;
+	};
+	limiters?: {
+		abortController?: AbortController;
+		maxSequentialErrors?: number;
+		maxModelCalls?: number;
+		tokenLimits?: TokenUsage;
+	};
+	finalReviewStep?: boolean;
+	validator?: (newContent: TreeNode) => void;
+	dumpDebugLog?: boolean;
+	planningStep?: boolean;
+}
+
+interface GenerateTreeEditsSuccessResponse {
+	status: "success";
+	tokenUsage: TokenUsage;
+}
+
+interface GenerateTreeEditsErrorResponse {
+	status: "failure" | "partial-failure";
+	errorMessage: "tokenLimitExceeded" | "tooManyErrors" | "tooManyModelCalls" | "aborted";
+	tokenUsage: TokenUsage;
+}
+
+/**
+ * Prompts the provided LLM client to generate valid tree edits.
+ * Applies those edits to the provided tree branch before returning.
+ *
+ * @remarks
+ * - Optional root nodes are not supported
+ * - Primitive root nodes are not supported
+ *
+ * @internal
+ */
+export async function generateTreeEdits<TSchema extends ImplicitFieldSchema>(
+	options: GenerateTreeEditsOptions<TSchema>,
+): Promise<GenerateTreeEditsSuccessResponse | GenerateTreeEditsErrorResponse> {
+	const idGenerator = new IdGenerator();
+	const editLog: EditLog = [];
+	let editCount = 0;
+	let sequentialErrorCount = 0;
+
+	const isRootNode = Tree.parent(options.treeNode) === undefined;
+	const simpleSchema = isRootNode
+		? getSimpleSchema(normalizeFieldSchema(options.treeView.schema).allowedTypes)
```

Comment:
I believe the following should work and is a bit simpler:
```suggestion
		? getSimpleSchema(options.treeView.schema)
```

Reply:
Has been removed as of commit fb37d5500e19de5dd5a5664513592c997115e358